rfc/src/rfcs/0006-unstable-api.md

```diff
@@ -0,0 +1,110 @@
+- **Feature Name:** Unstable APIs (unstable-api)
+- **RFC Tracking Issue**: <https://github.com/model-checking/kani/issues/2279>
+- **RFC PR:** <https://github.com/model-checking/kani/pull/2281>
+- **Status:** Under Review
+- **Version:** 0
+
+-------------------
+
+## Summary
+
+Provide a standard option for users to enable experimental APIs and features in Kani,
+and ensure that those APIs are off by default.
+
+## User Impact
+
+Add an opt-in model for users to try experimental APIs.
+The goal is to enable users to try features that aren't stable yet,
+which allow us to get valuable feedback during the development of new features and APIs.
+
+The opt-in model empower the users to control when some instability is acceptable,
+which makes Kani UX more consistent and safe.
+
+In order to reduce friction, we will also standardize how users opt-in to any Kani unstable feature.
+We will use similar syntax to the one used by the Rust compiler and Cargo.
+As part of this work, we will also deprecate and remove `--enable-unstable` option.
```

Comment:
The section doesn't explain the current approach to unstable options. I think it's worth explaining here for commenters who aren't familiar with it. That'll also make clearer what the benefits of this proposal are (UX standardization, an unstable feature catalog, etc.).